caom2/caom2/obs_reader_writer.py

```diff
@@ -2468,7 +2469,12 @@ def _add_element(self, name, text, parent):
         if isinstance(text, str):
             element.text = text
         else:
-            element.text = str(text)
+            if isinstance(text, float):
+                # in Python 2.7 str(float) might alter precision of float
+                # therefore call repr instead
+                element.text = repr(text)
+            else:
+                element.text = str(text)
```

Comment:
Maybe change the text argument to value to make it clear it could either be a string or number.

Reply:
Done.